### PR TITLE
Simplify incrementFileName

### LINKED ISSUE #55128: Making copy/paste filename incrementor switchable off
Issue Type: <b>Feature Request</b>

I use SQL schema patches where the file names start with the date to have them ordered and executed chronologically when automatically executing the migration on every release.

The file name auto-increment feature recognizes the first number of the file name (the year) and auto-increments it when I copy/paste an SQL patch to create a new one. This is unexpected an a bit dangerous as it can break the order of the migration. This screen shot shows the problem immediately:

wrong auto-increment

I would value it highly if the auto-increment feature for copy/pasted files could be turned off. (I assume some kind of 'smart' detection for dates or other patterns would be too error-prone to provide enough value.)

VS Code version: Code 1.25.1 (1dfc5e557209371715f655691b1235b6b26a06be, 2018-07-11T15:33:29.235Z)
OS version: Darwin x64 17.7.0

### PR DESCRIPTION
**Pull request by jeanp413**
Fixes #55128

### CODE REVIEW

File: src/vs/workbench/contrib/files/browser/fileActions.ts [hunk 1]
```diff
@@ -344,70 +344,32 @@ export function findValidPasteFileTarget(targetFolder: ExplorerItem, fileToPaste
...
-	if (isFolder && name.match(/^(\d+)/)) {
-		return name.replace(/^(\d+)(.*)$/, (match: string, ...groups: any[]) => {
-			let number = parseInt(groups[0]);
-			return number < maxNumber
-				? strings.pad(number + 1, groups[0].length) + groups[1]
-				: strings.format('{0}{1}.1', groups[0], groups[1]);
-		});
+	// name copy 5(.txt) => name copy 6(.txt)
+	// name copy(.txt) => name copy 2(.txt)
+	let suffixRegex = /^(.+ copy)( \d+)?$/;
```
Comment (isidorn) [inline src/vs/workbench/contrib/files/browser/fileActions.ts]: We prefer to use `const` over `let` when possible

File: src/vs/workbench/contrib/files/browser/fileActions.ts [hunk 2]
```diff
@@ -344,70 +344,32 @@ export function findValidPasteFileTarget(targetFolder: ExplorerItem, fileToPaste
...
-	if (isFolder && name.match(/(\d+)$/)) {
-		return name.replace(/(\d+)$/, (match: string, ...groups: any[]) => {
-			let number = parseInt(groups[0]);
-			return number < maxNumber
-				? strings.pad(number + 1, groups[0].length)
-				: strings.format('{0}.1', groups[0]);
-		});
+	let namePrefix = name;
+	let extSuffix = '';
+	if (!isFolder) {
```
Comment (isidorn) [inline src/vs/workbench/contrib/files/browser/fileActions.ts]: Would it help to use the paths.extname helper to simplify things here?

**Review by isidorn — APPROVED** [review]

**Comment by isidorn** [comment]
This is some great work. I love it, especailly the tests!
I was thinking if the differentiation between file and folder behavior is really necessery. But it seems that all native explorer have different behavior, so let's keep it like that. The corner case being I have a folder called "hello.t" -> "hello.t copy" instead of "hello copy.t". So let's keep the differentiation I think.

If you want you can address my two minor comments. I would merge this either way.

Thanks a lot!

**Comment by jeanp413** [comment]
@isidorn updated the PR, now I'm using `extname` and `basename`

**Comment by isidorn** [comment]
Great work, merging in.
Thanks a lot!

**Pull request merged**